### PR TITLE
Automation test for Curated groups and plots improvements 

### PR DESCRIPTION
**Pull request by labkey-sweta**
#### Rationale
Automation test for Curated groups and plots improvements - Active Filters

[Spec](https://docs.google.com/document/d/1v9Hk58ys0P9UVdWXYTvuYWL-0NMS994-Tgpy295l3YM/edit)

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

**Review by labkey-tchad — COMMENTED** [review]
This PR is a draft and the tests are failing. Are you looking for early feedback on any particular part?

**Comment by labkey-sweta** [comment]
> This PR is a draft and the tests are failing. Are you looking for early feedback on any particular part?

Yes, before i start refactoring more

File: test/src/org/labkey/test/pages/cds/GroupDetailsPage.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    public GroupDetailsPage(WebDriver driver)
+    {
+        super(driver);
+        waitForPage();
+    }
+
+    @Override
+    public void waitForPage()
+    {
+        _test.waitForText("Edit details");
```
Comment (labkey-tchad) [inline test/src/org/labkey/test/pages/cds/GroupDetailsPage.java]: `_test` is deprecated, don't use it. It's especially unnecessary here; `waitForText` is from `WebDriverWrapper`, which `LabKeyPage` extends.
```suggestion
       waitForText("Edit details");
```

File: test/src/org/labkey/test/pages/cds/GroupDetailsPage.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package org.labkey.test.pages.cds;
+
+import org.labkey.test.Locator;
+import org.labkey.test.pages.LabKeyPage;
+import org.labkey.test.util.Ext4Helper;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class GroupDetailsPage extends LabKeyPage<LabKeyPage.ElementCache>
```
Comment (labkey-tchad) [inline test/src/org/labkey/test/pages/cds/GroupDetailsPage.java]: Wrong generic here (probably why `elementCache()` wasn't working for you. These can be tricky so you should use the test file templates we have for IntelliJ (https://www.jetbrains.com/help/idea/settings-file-and-code-templates.html).
```suggestion
public class GroupDetailsPage extends LabKeyPage<GroupDetailsPage.ElementCache>
```

File: test/src/org/labkey/test/pages/cds/GroupDetailsPage.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+{
+    public GroupDetailsPage(WebDriver driver)
+    {
+        super(driver);
+        waitForPage();
+    }
```
Comment (labkey-tchad) [inline test/src/org/labkey/test/pages/cds/GroupDetailsPage.java]: `waitForPage` gets called automatically. Calling it here isn't necessary.
```suggestion
    public GroupDetailsPage(WebDriver driver)
    {
        super(driver);
    }
```

File: test/src/org/labkey/test/pages/cds/GroupDetailsPage.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Override
+    public void waitForPage()
+    {
+        _test.waitForText("Edit details");
+    }
+
+    public String getGroupName()
+    {
+        return newElementCache().groupName.getText();
```
Comment (labkey-tchad) [inline test/src/org/labkey/test/pages/cds/GroupDetailsPage.java]: Never call `newElementCache` directly.

File: test/src/org/labkey/test/pages/cds/GroupDetailsPage.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    public String getGroupDescription()
+    {
+        return newElementCache().groupDesc.getText();
+    }
+
+    public GroupDetailsPage deleteGroup(String option)
+    {
+        newElementCache().delete.click();
+        Ext4Helper.Locators.windowButton("Delete Group", option);
```
Comment (labkey-tchad) [inline test/src/org/labkey/test/pages/cds/GroupDetailsPage.java]: This is just a Locator, it doesn't do anything on its own.

**Review by labkey-tchad — CHANGES_REQUESTED** [review]

File: test/src/org/labkey/test/pages/cds/GroupDetailsPage.java (hunk 3 above)
Comment (labkey-sweta) [reply test/src/org/labkey/test/pages/cds/GroupDetailsPage.java]: done

**Pull request closed without merging**